themeServer.js

```diff
@@ -38,13 +41,35 @@ try {
   envJSON.THEME_SERVER = {
     host: `http://${address}`,
     port: `${PORT}`,
-    themePath: `${THEME_PATH}`
+    themePath: `${THEME_PATH}`,
+    imagePathStart: `${IMAGE_PATH}`,
+    includedImageFormats: INCLUDED_FORMATS
   }
   fs.writeFileSync(envFile, JSON.stringify(envJSON, null, 2))
 } catch (e) {
   console.log(e)
 }
 
+const findImagePaths = (
+  startPath,
+  includedFormats = ['.gif', '.png', '.jpeg', '.svg'],
+  result = {}
+) => {
+  if (!fs.existsSync(startPath)) return result
+  const filesDir = fs.readdirSync(startPath)
+  for (const file of filesDir) {
+    const filePath = startPath + '/' + file
+    if (!file.includes('.')) findImagePaths(filePath, includedFormats, result)
+    const fileExtensionIndex = file.indexOf('.')
+    const fileExtension = file.slice(fileExtensionIndex)
```

Comment:
Something you might want to consider. In the case of directories, there might not be a period (.), so `indexOf('.')` will return `-1`. Slice works with negative numbers, and uses a negative index by starting from the end. So an index of -1 may have the unintended consequence of getting a file extension of one letter.

Reply:
This is addressed by the line above where if the file does not have a period, the function calls itself.  It uses this to go deeper into each folder until it no longer finds anymore directories.

Reply:
Oh okay, got it, thanks! If that's the case, you may want to return the function instead of calling it: `if (!file.includes('.')) return findImagePaths(filePath, includedFormats, result)`. As it is right now, I think if it's a directory, it will call `findImagePaths` recursively on line 62, then once all those recursive calls are done, it will continue onto line 63. You might need to double-check both cases (with and without the return) to see which is the right one.

Reply:
Oh okay, got it, thanks! If that's the case, you may want to wrap lines 63-68 in an `else` block. I believe if the file does not have a period, it will call `findImagePaths` recursively on line 62, then continue onto line 63. You may want to double-check both cases (with and without the `else`) just to be sure.

Reply:
Or you could maybe do a check if the index of the period (.) is less than 0.

Reply:
You are correct that it continues to run after line 62.  Right now it compares the sliced extension to an array of allowed extensions, and since the weird slices from -1 are not included in the array, they are not used.  I'll adjust to make it so it doesn't continue after line 62 though as the current setup is a little unclear.

Reply:
I added a continue to the if statement.